core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1040,62 +1041,62 @@ object Parser extends ParserInstances {
     *  flatMap always has to allocate a parser, and the
     *  parser is less amenable to optimization
     */
-  def tailRecM[A, B](init: A)(fn: A => Parser[Either[A, B]]): Parser[B] =
+  def tailRecM[A, B](init: A)(fn: A => Parser0[Either[A, B]]): Parser0[B] =
```

Comment:
shouldn't this be def tailRecM0 to follow your other changes, and let tailRec be the 1 variant?